etl/naaccr_etl_sqlserver.sql

```diff
@@ -1334,9 +1330,186 @@ CREATE TABLE naaccr_data_points_temp
+                                      INNER JOIN concept AS c1                         ON cr1.concept_id_2 = c1.concept_id AND c1.vocabulary_id = 'NAACCR' AND c1.concept_class_id = 'NAACCR Value' AND c1.domain_id = 'Observation' AND c1.standard_concept = 'S'
+  							                      INNER JOIN naaccr_data_points ndp1               ON ndp.record_id = ndp1.record_id AND ndp1.naaccr_item_number = '390';
+
+
+  INSERT INTO observation
+  (
+      observation_id
+    , person_id
+    , observation_concept_id
+    , observation_date
+    , observation_datetime
+    , observation_type_concept_id
+    , value_as_number
+    , value_as_string
+    , value_as_concept_id
+    , qualifier_concept_id
+    , unit_concept_id
+    , provider_id
+    , visit_occurrence_id
+    , visit_detail_id
+    , observation_source_value
+    , observation_source_concept_id
+    , unit_source_value
+    , qualifier_source_value
+    -- , observation_event_id
+    -- , obs_event_field_concept_id
+    -- , value_as_datetime
+  )
+  SELECT
+      observation_id
+    , person_id
+    , observation_concept_id
+    , observation_date
+    , observation_datetime
+    , observation_type_concept_id
+    , value_as_number
+    , value_as_string
+    , value_as_concept_id
+    , qualifier_concept_id
+    , unit_concept_id
+    , provider_id
+    , visit_occurrence_id
+    , visit_detail_id
+    , observation_source_value
+    , observation_source_concept_id
+    , unit_source_value
+    , qualifier_source_value
+    -- , observation_event_id
+    -- , obs_event_field_concept_id
+    -- , value_as_datetime
+    FROM observation_temp;
+
+  IF OBJECT_ID('fact_relationship_temp', 'U') IS NOT NULL           -- Drop temp table if it exists
+    DROP TABLE fact_relationship_temp;
 
 
+  CREATE TEMPORARY TABLE fact_relationship_temp
+  (
+    domain_concept_id_1             INT           NOT NULL ,
+    fact_id_1                     	BIGINT        NOT NULL ,
+    domain_concept_id_2        		  INT           NOT NULL ,
+    fact_id_2              		   	  BIGINT        NOT NULL ,
+    relationship_concept_id         INT           NOT NULL ,
+    record_id                     	VARCHAR(255)  NULL
+  );
 
+  INSERT INTO fact_relationship_temp
+  (
+      domain_concept_id_1
+    , fact_id_1
+    , domain_concept_id_2
+    , fact_id_2
+    , relationship_concept_id
+    , record_id
+  )
+  SELECT
+      32527                             AS domain_concept_id_1			  -- Episode
+    , ep.episode_id                     AS fact_id_1
+    , 27                                AS domain_concept_id_2				-- Observation
+    , ob.observation_id                 AS fact_id_2
+    , 44818750                          AS relationship_concept_id  	-- Has occurrence
+    , NULL record_id
+   FROM episode_temp ep INNER JOIN observation_temp ob ON ep.person_id = ob.person_id AND ep.record_id = ob.record_id AND ep.episode_concept_id = 32528;			-- Disease First Occurrence
+
+
+  INSERT INTO fact_relationship
```

Comment:
If you run the ETL more than once, wouldn't the fact_relationship records from the previous runs still exist? If so we need to delete them beforehand.

Reply:
Make sense.